frontend/src/modules/automation/components/automation-form.vue

```diff
@@ -0,0 +1,324 @@
+import { required } from '@vuelidate/validators';
+import useVuelidate from '@vuelidate/core';
+import AppFormItem from '@/shared/form/form-item.vue';
+import AppNewActivityFilterOptions
+  from '@/modules/automation/components/filter-options/new-activity-filter-options.vue';
+import AppNewMemberFilterOptions from '@/modules/automation/components/filter-options/new-member-filter-options.vue';
+import AppAutomationWebhookAction from '@/modules/automation/components/action/webhook-action.vue';
+import AppAutomationSlackAction from '@/modules/automation/components/action/slack-action.vue';
+import { useAutomationStore } from '@/modules/automation/store';
+import Message from '@/shared/message/message';
+import { i18n } from '@/i18n';
+import formChangeDetector from '@/shared/form/form-change';
+
+const props = defineProps({
+  modelValue: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  type: {
+    type: String,
+    required: false,
+    default: 'webhook',
+  },
+  automation: {
+    type: Object,
+    required: false,
+    default: null,
+  },
+});
+
+const emit = defineEmits(['update:modelValue', 'update:automation']);
+
+const { createAutomation, updateAutomation } = useAutomationStore();
+
+const isDrawerOpen = computed({
+  get() {
+    return props.modelValue;
+  },
+  set() {
+    emit('update:modelValue', false);
+    emit('update:automation', null);
+  },
+});
+
+const type = computed(() => props.automation?.type || props.type);
+
+const isEdit = computed(() => props.automation !== null);
+
+const triggerOptions = ref([
+  {
+    label: 'New activity happened in your community',
+    value: 'new_activity',
+  },
+  {
+    label: 'New member joined your community',
+    value: 'new_member',
+  },
+]);
+
+const collapseOpen = ref('filterOptions');
+
+// Setup and validation
+const form = reactive({
+  name: '',
+  trigger: '',
+  settings: {},
+});
+
+const rules = {
+  trigger: {
+    required,
+  },
+};
+
+const $v = useVuelidate(rules, form);
+
+const { hasFormChanged, formSnapshot } = formChangeDetector(form);
+
+// Prefill form
+const fillForm = (automation) => {
+  form.name = automation.name || '';
+  form.trigger = automation.trigger || '';
+  form.settings = automation.settings || {};
+  formSnapshot();
+};
+
+const reset = () => {
+  form.name = '';
+  form.trigger = '';
+  form.settings = {};
+  formSnapshot();
+};
+
+watch(() => props.modelValue, () => {
+  if (type.value && props.automation) {
+    fillForm(props.automation);
+  } else {
+    reset();
+  }
+});
+
+// Submit form
+const sending = ref(false);
+
+const doSubmit = () => {
+  if ($v.value.$invalid) {
+    return;
+  }
+  sending.value = true;
+  const data = {
+    name: form.name ?? i18n(`entities.automation.triggers.${form.trigger}`),
+    type: type.value,
+    trigger: form.trigger,
+    settings: {
+      ...form.settings,
+    },
+  };
+  if (!isEdit.value) {
+    createAutomation(data)
```

Comment:
After creating an automation should the list of automations be fetched again?
I tried to create an automation and I had to refresh the page to see it on the automations list

This happens when the empty state is shown